infra-parent/infra-core/src/main/java/com/redhat/service/smartevents/infra/core/exceptions/ErrorHrefVersionBuilderImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.redhat.service.smartevents.infra.core.exceptions;
+
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.validation.ConstraintViolation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class ErrorHrefVersionBuilderImpl implements ErrorHrefVersionBuilder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ErrorHrefVersionBuilderImpl.class);
+
+    private Instance<ErrorHrefVersionProvider> builders;
+
+    protected ErrorHrefVersionBuilderImpl() {
+        //CDI proxy
+    }
+
+    @Inject
+    public ErrorHrefVersionBuilderImpl(@Any Instance<ErrorHrefVersionProvider> builders) {
+        this.builders = builders;
+    }
+
+    @Override
+    public String buildHref(Throwable e, String id) {
+        if (e instanceof HasErrorHref) {
```

Comment:
Explicit RHOSE `Exceptions` have an explicit property for the `href`.

We now only use the `ErrorHrefVersionProvider` as a fallback mechanism for the generic exceptions `ExternalUserException` and `InternalPlatformException`. These need to be common to all versions as our single-instance/cross-version `ExceptionMapper` implementations need to use a common class.